specs/TryUpdateRuntime.md

```diff
@@ -0,0 +1,225 @@
+{
+    if (IsCurrentVersionSameOrNewer(minimalVersionRequired))
+    {
+      callback(true);
+      return;
+    }
+    auto options = Microsoft::WRL::Make<CoreWebView2EnvironmentOptions>();
+    CreateCoreWebView2EnvironmentWithOptions(nullptr, nullptr, options.Get(),
+        Callback<ICoreWebView2CreateCoreWebView2EnvironmentCompletedHandler>(
+            [callback, minimalVersionRequired](HRESULT result, ICoreWebView2Environment* environment) -> HRESULT {
+                wil::com_ptr<ICoreWebView2Environment> webViewEnvironment = environment;
+                auto experimentalEnvironment3 =
+                webViewEnvironment.try_query<ICoreWebView2ExperimentalEnvironment3>();
+                HRESULT hr = experimentalEnvironment3->TryUpdateRuntime(
+                    Callback<ICoreWebView2ExperimentalTryUpdateRuntimeCompletedHandler>(
+                        [callback, minimalVersionRequired, experimentalEnvironment3](HRESULT errorCode,
+                           ICoreWebView2ExperimentalUpdateRuntimeResult* result) -> HRESULT {
+                            COREWEBVIEW2_RUNTIME_UPDATE_STATUS updateStatus =
+                                COREWEBVIEW2_RUNTIME_UPDATE_STATUS_FAILED;
+                            if ((errorCode == S_OK) && result)
+                            {
+                              CHECK_FAILURE(result->get_UpdateRuntimeStatus(&updateStatus));
+                            }
+                            if (updateStatus != COREWEBVIEW2_RUNTIME_UPDATE_STATUS_UPDATED)
+                            {
+                              callback(false);
+                            }
+                            else
+                            {
+                              callback(IsCurrentVersionSameOrNewer(minimalVersionRequired));
+                            }
+                            return S_OK;
+                        })
+                        .Get());
+                return S_OK;
+            })
+            .Get());
+}
+
+```
+
+# Remarks
+Try to update installed Microsoft Edge WebView2 Runtime.
+This will potentially result in a new version of the Edge Webview2
+Runtime being installed and `NewBrowserVersionAvailable` event fired.
+There is no guarantee on the order of that event being fired and
+TryUpdateRuntimeis completed handler being invoked. Besides the
+`NewBrowserVersionAvailable` event, there will be no impact to any
+currently running WebViews when update is installed.
+The latest version can always be queried using `GetAvailableCoreWebView2BrowserVersionString` API.
+This is only supported for installed Edge WebView2 Runtime. When running
+fixed version Edge WebView2 Runtime or non stable channel Edge browser,
+this API will return `HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED)`.
+There could only be one active TryUpdateRuntime operation, calling the API
+before completed handler for previous call is invoked will fail with
+`HRESULT_FROM_WIN32(ERROR_BUSY)`.
```

Comment:
That would be complicated, we will have to track which thread each call comes from and report back on those thread, and ensure that we don't lost the completed handler in some race condition. As it is likely a bug for an app to call the API multiple times, it would be better and simple to reject the call immediately.